crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -181,18 +183,22 @@ pub struct TrustpayMandatoryParams {
     pub billing_postcode: Secret<String>,
 }
 
-fn get_trustpay_payment_method(bank_redirection_data: &BankRedirectData) -> TrustpayPaymentMethod {
-    match bank_redirection_data {
-        api_models::payments::BankRedirectData::Giropay { .. } => TrustpayPaymentMethod::Giropay,
-        api_models::payments::BankRedirectData::Eps { .. } => TrustpayPaymentMethod::Eps,
-        api_models::payments::BankRedirectData::Ideal { .. } => TrustpayPaymentMethod::IDeal,
-        api_models::payments::BankRedirectData::Sofort { .. } => TrustpayPaymentMethod::Sofort,
+impl TryFrom<&BankRedirectData> for TrustpayPaymentMethod {
+    type Error = Error;
+    fn try_from(value: &BankRedirectData) -> Result<Self, Self::Error> {
+        match value {
+            api_models::payments::BankRedirectData::Giropay { .. } => Ok(Self::Giropay),
+            api_models::payments::BankRedirectData::Eps { .. } => Ok(Self::Eps),
+            api_models::payments::BankRedirectData::Ideal { .. } => Ok(Self::IDeal),
+            api_models::payments::BankRedirectData::Sofort { .. } => Ok(Self::Sofort),
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
```

Comment:
We should log the payment method for better debugging purpose

Reply:
resolving, as this is not required.